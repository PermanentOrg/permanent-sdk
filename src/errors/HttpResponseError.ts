export class HttpResponseError extends Error {
  public readonly statusCode: number;

  public constructor(statusCode: number, message: string) {
    super(message);
    this.statusCode = statusCode;
  }
}
