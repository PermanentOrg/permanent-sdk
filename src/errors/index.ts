export * from './HttpResponseError';
export * from './ValidationError';
