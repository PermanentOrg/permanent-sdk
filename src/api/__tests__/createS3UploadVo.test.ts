import nock from "nock";
import { createS3UploadVo } from "..";
import { ValidationError } from "../../errors";

describe("createS3UploadVo", () => {
	it("should create a presigned post", async () => {
		nock("https://permanent.local")
			.post("/api/record/getPresignedUrl", {
				displayName: "testFile.txt",
				parentFolderId: 1,
				uploadFileName: "testFile",
				fileType: "text/plain",
				size: 12,
			})
			.replyWithFile(
				200,
				`${__dirname}/fixtures/createS3UploadVo/s3Upload.json`,
				{
					"Content-Type": "application/json",
				},
			);

		const s3UploadVo = await createS3UploadVo(
			{
				bearerToken: "12345",
				baseUrl: "https://permanent.local/api",
			},
			{
				displayName: "testFile.txt",
				parentFolderId: 1,
				uploadFileName: "testFile",
				fileType: "text/plain",
				size: 12,
			},
		);

		expect(s3UploadVo).toMatchSnapshot();
	});

	it("should error when provided an invalid response", async () => {
		nock("https://permanent.local")
			.post("/api/record/getPresignedUrl", {
				displayName: "testFile.txt",
				parentFolderId: 1,
				uploadFileName: "testFile",
				fileType: "text/plain",
				size: 12,
			})
			.reply(200, '["Not at all a s3UploadVo"]', {
				"Content-Type": "application/json",
			});

		await expect(
			createS3UploadVo(
				{
					bearerToken: "12345",
					baseUrl: "https://permanent.local/api",
				},
				{
					displayName: "testFile.txt",
					parentFolderId: 1,
					uploadFileName: "testFile",
					fileType: "text/plain",
					size: 12,
				},
			),
		).rejects.toBeInstanceOf(ValidationError);
	});
});
