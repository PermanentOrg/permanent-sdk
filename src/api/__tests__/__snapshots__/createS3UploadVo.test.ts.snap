// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`createS3UploadVo should create a presigned post 1`] = `
{
  "destinationUrl": "https://permanent-local.s3.amazonaws.com/_slifty/unprocessed/12345678-e6ef-4238-85d3-20001abebb63",
  "presignedPost": {
    "fields": {
      "Key": "_slifty/unprocessed/12345678-e6ef-4238-85d3-20001abebb63",
      "Policy": "123456789=",
      "X-Amz-Algorithm": "AWS4-HMAC-SHA256",
      "X-Amz-Credential": "12345678/20230123/us-west-2/s3/aws4_request",
      "X-Amz-Date": "20230123T210239Z",
      "X-Amz-Signature": "1234567890",
      "bucket": "permanent-local",
    },
    "url": "https://s3.us-west-2.amazonaws.com/permanent-local",
  },
}
`;
